client/sources/ok_test/models.py

```diff
@@ -100,9 +100,11 @@ def unlock(self, interact):
                                 for case in suite.cases])
         for suite_num, suite in enumerate(self.suites):
             for case_num, case in enumerate(suite.cases):
+                case_id = '{} > Suite {} > Case {}'.format(
```

Comment:
I think we shouldn't hash; just use numbered IDs. Yes, some student could
corrupt the numbering, but some student could also fabricate a whole
transcript if she wanted to.... both seem very unlikely.

On Mon, Jun 29, 2015 at 3:06 PM, albert12132 <notifications@github.com>
wrote:

> In client/sources/ok_test/models.py
> <https://github.com/Cal-CS-61A-Staff/ok-client/pull/97#discussion_r33519623>
> :
>
> > @@ -100,9 +100,11 @@ def unlock(self, interact):
> >                                  for case in suite.cases])
> >          for suite_num, suite in enumerate(self.suites):
> >              for case_num, case in enumerate(suite.cases):
> > +                case_id = '{} > Suite {} > Case {}'.format(
>
> The data format described in the doc looks correct to me (for what we've
> currently agreed on).
>
> @soumya <https://github.com/soumya>, those two cases would record the
> following 4 attempts:
>
> {'prompt': '>>> x = 5\n>>>square(x)',   # first prompt of first case
>  'answer': '25'},
> {'prompt': '>>> square(-x)',   # second prompt of first case
>  'answer': '25'},
> {'prompt': '>>> x = -5\n>>>square(x)',   # first prompt of second case
>  'answer': '25'},
> {'prompt': '>>> square(-x)',   # second prompt of second case
>  'answer': '25'},
>
> You'll notice that attempt 2 and attempt 4 are identical; this occurs
> because attempts are recorded for each prompt within a test case (attempts
> are not recorded for the test case over all).
>
> @ksteph <https://github.com/ksteph> your stated purpose of the hash
> aligns with my understanding of it. From what @soumyabasu
> <https://github.com/soumyabasu> described, where ID =
> hash(assignment_name, test_name, prompt) is insufficient, because it's
> possible for the same prompt to show up more than once in a test (see
> example I gave above).
>
> My point about removing the case_id is that there's no reason to have
> both a case_id and a hash, since they're supposed to do the same thing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cal-CS-61A-Staff/ok-client/pull/97/files#r33519623>.
>